Added missing package prereqs to installation guide.

jaedoucette opened the pull request:
Added npm and requirement for more modern docker-compose.

mlippert commented on docs/development-guide.md:
```diff
@@ -24,6 +24,8 @@ this relative directory layout.
 - docker
 - make
 - Python 3 (only for needed for deploying)
+- npm
```
I don't think `npm` is really _required_ because the only uses of it can be done inside of docker containers that have the correct version of node installed. And I don't think I'd say `npm` in any case I'd probably say `node`. But having `node 10` or probably shortly `node 12` installed is certainly a nice to have (https://nodejs.org/en/download/).

jaedoucette replied:
At one point in the  process, an "npm install" needs to be run locally in the signalmaster directory, rather than inside a container. Changing it to node makes sense though.

mlippert replied:
Huh, I forgot it still said that. It's no longer true (I think I updated the doc in edu-docker but didn't make the changes here).
Mentioning it as not required but nice to have is probably a good idea.

jaedoucette replied:
Ok. Sounds like I should I also remove the npm install  instructions for signalmaster.

mlippert commented on docs/development-guide.md:
```diff
@@ -24,6 +24,8 @@ this relative directory layout.
 - make
 - Python 3 (only for needed for deploying)
+- npm
+- docker-compose, with a more modern version than Debian's defaults (https://docs.docker.com/compose/install/)
```
yes, good point, `docker-compose` has to be installed separately from `docker` and really should be in this list. I don't think I'd mention Debian here, since this should be OS agnostic.
OS specific sections may be warranted at some point.

jaedoucette replied:
Good point. I was unable to determine the maximum version of docker's YML that we're using, only that the one in Mint/Ubuntu/Debian's repo is very far out of date. If you know the maximum YML version we're using, I can translate that into a minimum docker-compose version.

mlippert replied:
Actually I think I've set the version in the docker-compose.yml files. `version: '3.6'` is what i see that I've specified there

jaedoucette replied:
Perfect. I'll add the version 18.02.0+ for docker-compose.

mlippert commented on docs/development-guide.md:
````diff
@@ -38,13 +40,7 @@ cd riff-docker
-npm install
-cd ../riff-docker
-```
+
````
yes but signalmaster still needs to be inited. see https://github.com/rifflearning/edu-docker/blob/master/docs/development-guide.md

basically add `make init-signalmaster` to the section above.

jaedoucette replied:
Makes sense. I added ti to the end of the first command block.

jaedoucette committed:
Update development-guide.md

- Updated prereqs
- Added version 18.02.0+ to docker compose
- Added node 10 prereq as "recommended"
- Replace the npm install of signalmaster with `make init-signalmaster`

mlippert commented:
I've squashed the commits into a single commit (hmm I could have done that w/ the merge button, well maybe I'll try that next time :-) )
